merkle/hstar2_test.go

```diff
@@ -25,6 +24,8 @@ import (
 	"github.com/google/trillian/testonly"
 )
 
+const treeID = int64(0)
```

Comment:
I'm not exactly sure what to do here. The maphasher - which matches the Python / C++ merkle tree code - does not use TreeID and result in different hashes.  

We could 
- Change the maphasher to include TreeID and generate new test vectors. 
- Keep the existing ones and add a new test with CONIKS test vectors. But do we really want to make the merkle package dependent on all the sub hashing algorithms?